code/modules/food_and_drinks/kitchen_machinery/food_replicator.dm

```diff
@@ -0,0 +1,631 @@
+		return 0
+	if(cost > food_paste)
+		visible_message("<span class='warning'>Scans indicate that the replicator lacks sufficient food paste to replicate this dish. Please recycle more food or insert a less complex dish.</span>")
+		return 0
+	operating = 1
+	playsound(loc, 'sound/misc/interference.ogg', 50, 1)
+	var/path_to_dupe = loaded.type
+	var/obj/item/weapon/reagent_containers/food/snacks/dupe = new path_to_dupe()
+	dupe.name = loaded.name		//we even replicate renamed dishes' names!
+	dupe.duped = 1
+	replicate_reagents(loaded.reagents.reagent_list, dupe)
+	handle_emag_effect(dupe)
+	sleep(20)
+	adjust_paste(-cost)
+	use_power(active_power_usage)
+	dupe.forceMove(get_turf(src))
+	operating = 0
+	playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+	return 1
+
+//copy_food_from_memory is used to create a replication of a previously saved food item from the given memory slot
+/obj/machinery/food_replicator/proc/copy_food_from_memory(slot = 0)
+	var/food_item
+	var/dish_name
+	var/list/reagent_memory = list()
+	switch(slot)
+		if(1)
+			dish_name = memory1["name"]
+			food_item = memory1["item"]
+			reagent_memory = memory1["reagents"]
+		if(2)
+			dish_name = memory2["name"]
+			food_item = memory2["item"]
+			reagent_memory = memory2["reagents"]
+		if(3)
+			dish_name = memory3["name"]
+			food_item = memory3["item"]
+			reagent_memory = memory3["reagents"]
+		if(4)
+			dish_name = memory4["name"]
+			food_item = memory4["item"]
+			reagent_memory = memory4["reagents"]
+		else
+			visible_message("<span class='warning'>ERROR 404: MEMORY ADDRESS NOT FOUND!</span>")
+			return 0
+	if(!food_item)
+		visible_message("<span class='warning'>Memory slot [slot] appears to lack a saved food item. Alert a coder if something is saved in that slot!</span>")
+		return 0
+	var/cost = get_cost_from_memory(reagent_memory)
+	if(cost == -1)
+		visible_message("<span class='warning'>There was an error while calculating necessary resources, aborting replication.</span>")
+		return 0
+	if(!cost)
+		visible_message("<span class='warning'>Scans indicate that the replicator is currently unable to replicate this dish due to hardware constraints.</span>")
+		return 0
+	if(cost > food_paste)
+		visible_message("<span class='warning'>Scans indicate that the replicator lacks sufficient food paste to replicate this dish. Please recycle more food or select a less complex dish.</span>")
+		return 0
+	operating = 1
+	playsound(loc, 'sound/misc/interference.ogg', 50, 1)
+	var/obj/item/weapon/reagent_containers/food/snacks/dupe = new food_item()
+	dupe.name = dish_name		//we even replicate renamed dishes' names!
+	dupe.duped = 1
+	replicate_reagents(reagent_memory, dupe)
+	handle_emag_effect(dupe)
+	sleep(20)
+	adjust_paste(-cost)
+	use_power(active_power_usage)
+	dupe.forceMove(get_turf(src))
+	operating = 0
+	playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+	return 1
```

Comment:
Why I didn't notice this and make a proc to handle instead of copying the code is baffling to even myself at this point.

I'll be honest, a lot of this code was at night after work, so I probably wasn't in my brightest state.